app/controllers/books_controller.rb

```diff
@@ -3,7 +3,10 @@ class BooksController < ApplicationController
   before_action :find_book, only: [:edit, :update, :destroy]
 
   def index
+    @search_term = params.dig(:search, :term)
+    search_term_with_wildcards = "%#{@search_term}%"
     @books = Book.ordered.page(params[:page])
+      .where("name LIKE ? OR isbn LIKE ?", search_term_with_wildcards, search_term_with_wildcards)
```

Comment:
The next step for this search would be the add it to the model as a scope.

You can [read about scopes in the Rails documentation](https://guides.rubyonrails.org/active_record_querying.html#scopes).

We want to get this functionality out of the controller and into the model so the controller only needs to know about an abstraction of searching and not the specific details of how the search is performed.

With the implementation in the model, we now have a concept of what it means to perform a generic search on a `Book`, which can  be used elsewhere.

It also allows us to easily unit test the new `search` scope.

We should end up with `Book.ordered.search(@search_term).page(params[:page])` in the controller.

The model's `search` scope would take the raw search term, decorate it with wildcards, and pass it into the `where`.

If you do the same for the users_controller below, you might notice that both models will need to know how to decorate a search term with wildcards. With that in mind you might like to create a method on `ApplicationRecord` that can be used in both scopes.

Reply:
Similarly, both controllers paginate a collection. You can create encapsulate that process in a method `paginate` on your `ApplicationController`.

So in `books_controller.rb` you'd end up with:

```ruby
paginate(Book.ordered.search(@search_term))
```

In `ApplicationController` you'd need to make a method called `paginate` that accepts a collection as an argument and can call the `page` method on it, the same as you have done in this controller.

The reason you'd do this is to remove the need for this controller to know about how pagination is performed.

This is a good idea because:
1. It makes pagination in the controller slightly easier to perform (because you don't have to remember what the page param is called)
2. It makes it easier to change pagination tool, because the implementation detail of paginating a collection is defined in one place (`ApplicationController`)